src/component/DetectionComponent.ts

```diff
@@ -1,127 +1,109 @@
 /// <reference path="../../typings/browser.d.ts" />
 
 import * as rx from "rx";
+import * as vd from "virtual-dom";
 
 import {Node} from "../Graph";
 import {Container, Navigator} from "../Viewer";
-import {APIv2, IRect} from "../API";
+import {APIv3} from "../API";
 
 import {ComponentService, Component} from "../Component";
+import {IVNodeHash} from "../Render";
+
+interface IRect {
+    rect: number[];
+    score: string;
+    value: string;
+    object: string;
+    key: string;
+    package: string;
+}
 
 export class DetectionComponent extends Component {
     public static componentName: string = "detection";
     private _disposable: rx.IDisposable;
-
-    private rectContainer: HTMLElement;
-    private detectionData: any;
-    private apiV2: APIv2;
+    private apiV3: APIv3;
 
     constructor(name: string, container: Container, navigator: Navigator) {
         super(name, container, navigator);
-        this.apiV2 = navigator.apiV2;
+        this.apiV3 = navigator.apiV3;
     }
 
     protected _activate(): void {
-        let child: HTMLElement = document.createElement("div");
-        child.className = "rectContainer";
-
-        this.rectContainer = child;
-        this._container.element.appendChild(this.rectContainer);
-
-        this._disposable = this._navigator
-            .stateService
-            .currentNode$.subscribe((node: Node): void => {
-                this.setRectContainer(node.image.width, node.image.height);
-                this.removeRectsFromDOM();
-
-                this.apiV2.im.callOr(node.key).then((data: any) => {
-                    this.detectionData = data;
-                    this.updateRects(this.detectionData.or_rects);
-                });
-            });
+        this._disposable = this._navigator.stateService.currentNode$.flatMap((node: Node): rx.Observable<any> => {
+            let path: string = "['imageByKey']";
+            path += "['" + node.key + "']";
+            path += "['ors'][0..20]['key', 'obj', 'rect', 'value', 'package', 'score']";
+
+            return this.apiV3.model.get(path);
+        }).map((ors: any): IVNodeHash => {
+            let rects: IRect[] = [];
+            delete ors.json.imageByKey.$__path;
```

Comment:
Enforcing the user of the Falcor API to delete this property every time seems like a possible cause of future bugs. Can this be encapsulated in the APIv3 in any way?